source/dofs/dof_tools_constraints.cc

```diff
@@ -2594,19 +2593,18 @@ namespace DoFTools
               const unsigned int local_parameter_dof
                 = coarse_fe.system_to_component_index(local_dof).second;
 
-              copy_data.global_parameter_representation[local_dof] = 0.;
+              copy_data.global_parameter_representation[local_parameter_dof] = 0.;
 
               // distribute the representation of
               // @p{local_parameter_dof} on the parameter grid cell
               // @p{cell} to the global data space
               coarse_to_fine_grid_map[cell]->
               set_dof_values_by_interpolation (parameter_dofs[local_parameter_dof],
-                                               copy_data.global_parameter_representation[pos]);
+                                               copy_data.global_parameter_representation[local_parameter_dof]);
 
 #ifdef DEAL_II_WITH_MPI
-              copy_data.global_parameter_representation[pos].update_ghost_values ();
+              copy_data.global_parameter_representation[local_parameter_dof].update_ghost_values ();
```

Comment:
The patch looks good and is ready to merge. I guess this is fixing a fairly recent problem, if it existed in r8.3 an entry to doc/news/changes.h should be added.

However, this line caught my attention. The code is calling "update_ghost_values()" very far inside a loop over all cells (run by WorkStream) and local dofs. This is not the right place to do because update_ghost_values() requires communication. If two processors have a different number of cells (it appears that the code is using parallel distributed triangulations where this is an issue), the code will lock because not all processors will call update_ghost_values() the same number of times. Also, calling update_ghost_values inside a multithreaded environment (WorkStream) is not recommended (even though it should work due to a mutex).

I must admit that I do not fully understood this function, in particular why there is a need for a global data structure structure covering all degrees of freedom, given that one finally only fills an std::map in copy_intergrid_weights, and why there is a need for calling update_ghost_values() while working on cells... It seems to me that filling the parallel vector for all cells (without ever updating the ghost values) and then filling the weights in a second loop would be a more correct way to do things, but again I don't really understand what the code is doing.